fuse-products/src/main/java/software/tnb/product/csb/application/TomcatSpringBootApp.java

```diff
@@ -0,0 +1,194 @@
+        }
+
+        Path logFile = getLogPath();
+        // startup.sh starts on another process, let's use catalina run so that we can control the lifecyle
+        ProcessBuilder processBuilder = new ProcessBuilder(tomcatHome + File.separator + "bin" + File.separator + "catalina.sh", "run")
+            .redirectError(logFile.toFile())
+            .redirectOutput(logFile.toFile());
+
+        try {
+            tomcatProcess = processBuilder.start();
+            Runtime.getRuntime().addShutdownHook(new Thread(() -> {
+                if (tomcatProcess != null) {
+                    tomcatProcess.destroyForcibly();
+                }
+            }));
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+
+        LOG.info("Starting tomcat in {}", tomcatHome);
+    }
+
+    private void downloadTomcat() {
+        try {
+            tomcatTmpDirectory = Files.createTempDirectory("tnb-tomcat");
+            LOG.info("Downloading tomcat in {}", tomcatTmpDirectory);
+            FileUtils.copyURLToFile(new URL(SpringBootConfiguration.tomcatZipUrl()), tomcatTmpDirectory.resolve(TOMCAT_ARCHIVE_NAME).toFile());
+
+            ZipUtils.unzip(tomcatTmpDirectory.resolve(TOMCAT_ARCHIVE_NAME), tomcatTmpDirectory.resolve(TOMCAT_PARENT_DIRECTORY));
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Override
+    public void start() {
+        startTomcat();
+
+        // replace packaging and extend SpringBootServletInitializer
+        try {
+            Path pomPath = TestConfiguration.appLocation().resolve(name).resolve("pom.xml");
+            String pom = Files.readString(pomPath);
+            pom = pom.replace("<artifactId>" + name + "</artifactId>",
+                "<artifactId>" + name + "</artifactId><packaging>war</packaging>");
+            Files.write(pomPath, pom.getBytes(StandardCharsets.UTF_8));
+
+            Path mainPath = TestConfiguration.appLocation().resolve(name)
+                .resolve("src")
+                .resolve("main")
+                .resolve("java")
+                .resolve("com")
+                .resolve("test")
+                .resolve("MySpringBootApplication.java");
+            String main = Files.readString(mainPath);
+
+            main = main.replace("class MySpringBootApplication {",
+                """
+                    class MySpringBootApplication extends org.springframework.boot.web.servlet.support.SpringBootServletInitializer {
+                    
+                        @Override
+                        protected org.springframework.boot.builder.SpringApplicationBuilder 
+                                configure(org.springframework.boot.builder.SpringApplicationBuilder application) {
+                            return application.sources(MySpringBootApplication.class);
+                        }
+                    """);
```

Comment:
if this is something that should be done for all tomcat apps, it should be done by a `TomcatCustomizer`